### PR TITLE
feat: custom user agent Auth changes for UI handoff

### PR DESCRIPTION
**Pull request by erinleigh90**
#### Description of changes
Creates and exports InternalAuth class from /internals scope with added customUserAgentDetails parameter on public API's.

NOTES
- To see the changes to InternalAuth after moving the changes over from Auth, see [this comparison](https://github.com/erinleigh90/amplify-js/compare/cfe55be..164290a#diff-d6abeead864c341b445fb8f3021ac370aec8e25133d51edd99641df7c1284b68)

#### Description of how you validated changes
- [x] unit tests passing
- [x] integ tests passing
- [x] smoke-tested in sample app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by stocaaro — DISMISSED** [review]
Got my [version diff](https://github.com/aws-amplify/amplify-js/pull/11606/files/cfe55be8a75c90eced61ffd43cf90f841992fc15..01d5c2e10eecff0b6ef8b07a5d76477112b4753e#diff-d6abeead864c341b445fb8f3021ac370aec8e25133d51edd99641df7c1284b68) figured on this one.

LGTM

File: packages/auth/src/internals/InternalAuth.ts [hunk 1]
```diff
@@ -0,0 +1,2872 @@
...
+	 * @return - A promise resolves callback data if success
+	 */
+	public signUp(
+		params: string | SignUpParams,
+		...restOfAttrs: string[]
+	): Promise<ISignUpResult> {
+		return this.internalSignUp(params, restOfAttrs);
+	}
+
+	public internalSignUp(
```
Comment (jimblanc) [inline packages/auth/src/internals/InternalAuth.ts:324]: Won't this API be exposed on the customer facing `Auth` class (which extends from this) as well?

**Review by jimblanc — CHANGES_REQUESTED** [review]
Need to confirm the visibility of `internalSignUp` on the customer-facing class, otherwise looks okay

File: packages/auth/src/internals/InternalAuth.ts (hunk 1 above)
Comment (erinleigh90) [reply packages/auth/src/internals/InternalAuth.ts]: Yes, you are right, thank you for catching!

**Review by jimblanc — DISMISSED** [review]
Looks good!

**Review by elorzafe — DISMISSED** [review]
Looks good!

Thanks @erinleigh90 🥇

**Review by elorzafe — APPROVED** [review]

**Review by jimblanc — DISMISSED** [review]

**Review by elorzafe — APPROVED** [review]

**Review by david-mcafee — APPROVED** [review]
Approving bundle size limit increases on behalf of the Data team.

**Review by jimblanc — APPROVED** [review]

**Pull request merged**